module-syncing/README.md

```diff
@@ -0,0 +1,144 @@
+
+##### All modules dropped a puppet version (this is more important the a feature label)
+
+Run the following command, replacing `X`:
+
+```bash
+bundle exec rake 'git:create_pr[Dropped Puppet X, sync to current standards, backwards-incompatible]'
+```
+
+##### All modules added a puppet version
+
+Run the following command, replacing `X`:
+
+```bash
+bundle exec rake 'git:create_pr[Added Puppet X, sync to current standards, feature]'
+```
+
+##### Most modules were unchanged with regards to the puppet version
+
+Run the following command:
+
+```bash
+bundle exec rake 'git:create_pr[Sync to current standards, maintenance]'
+```
+
+After running this command you will need to go to each module on GitHub that did have a change in its puppet requirements and:
+
+- replace the maintenance label
+- update the title of the pull request to either `Dropped Puppet X, sync to current standards` or `Added Puppet X, sync to current standards` as needed (be sure to replace the `X`)
+
+#### GH Actions added or removed
+
+If you have added or removed a GitHub action to a repo then you will need to do the following instead of bulk-creating pull requests:
+
+1. ensure you have [Hub](https://hub.github.com) installed
+2. cd to the first module repository
+3. run the command below to create a pull request based on whcih case fits the module (note that any module that had its puppet version requirements changed will need a label of either `backwards-incompatible` or `feature` instead of `maintenance` based on if a version was dropped or added):
+   - no additions or removals: `hub pull-request --push --browse --labels maintenance`. If the version of puppet required changed, please see the last section for how to modify the pull request title hub prompts you for.
+   - added or removed actions:
+     1. if adding an action, temporarily change it to run on push instead of just on pull requests so that its initial run will be kicked off. Add this change as an additional commit so we can easily remove it later. Skip to step 2 if not adding an action.
+     2. run `hub pull-request --push --browse --labels maintenance --draft`. If the version of puppet required changed, please see the last section for how to modify the pull request title hub prompts you for.
+     3. wait for the initial CI run to finish
+     4. go to the repository setting on GitHub and edit the branch protection rule for main so that it includes only the desired tests (add new ones and/or remove unneeded ones)
+     5. if a commit was added in step 1 and then execute `git reset --hard HEAD~1` followed by `git push -f`
+     6. if a push was done via step 5, wait for CI to complete again
+     7. mark the PR as ready for review on GitHub (aka take it out of draft mode)
+4. repeat these steps for each additional module. I suggest working on the modules in alphabetical order to simplify keeping track of which one is next.
+
+## Bulk releases
+
+TODO: fill this in
```

Comment:
As a side note here, since we use the on workflow dispatch for release, it should be possible to use `gh workflow run` to run the release workflow https://cli.github.com/manual/gh_actions

Reply:
nice!